kivy/core/window/window_sdl2.py

```diff
@@ -514,6 +514,10 @@ def mainloop(self):
             if action == 'dropfile':
                 dropfile = args
                 self.dispatch('on_dropfile', dropfile[0])
+            # A quit event might be received while the app is paused
+            elif action == 'quit':
+                EventLoop.quit = True
```

Comment:
```suggestion
                if self.dispatch('on_request_close'):
                    continue
                EventLoop.quit = True
```

Reply:
@matham, I was thinking about this. My concern is if the user will return `True` from `on_request_close`, it will hang.

Reply:
Yes, and I agree, but I'm wondering if that would introduce an inconsistency. Because why can a user ignore a quit event when *not* paused without it hanging? Or maybe it would also hang on android (it doesn't on desktop)?

Could you test whether returning True from `on_request_close` hangs the app if it's not paused? Or maybe android always pauses an app before closing it (I'm not super familiar with android)?

Reply:
> Could you test whether returning True from on_request_close hangs the app if it's not paused?

Sure.

Reply:
@matham, I did some tests and it appeared that returning True from on_request_close **does not hangs** the app if it's not paused. Then I continued my investigation and came to libsdl's SDL_android.d:
```c
/* Quit */
JNIEXPORT void JNICALL SDL_JAVA_INTERFACE(nativeQuit)(
                                    JNIEnv* env, jclass cls)
{
    /* Discard previous events. The user should have handled state storage
     * in SDL_APP_WILLENTERBACKGROUND. After nativeQuit() is called, no
     * events other than SDL_QUIT and SDL_APP_TERMINATING should fire */
    SDL_FlushEvents(SDL_FIRSTEVENT, SDL_LASTEVENT);
    /* Inject a SDL_QUIT event */
    SDL_SendQuit();
    SDL_SendAppEvent(SDL_APP_TERMINATING);
    /* Resume the event loop so that the app can catch SDL_QUIT which
     * should now be the top event in the event queue. */
    if (!SDL_SemValue(Android_ResumeSem)) SDL_SemPost(Android_ResumeSem);
}
```

I thought that the quit event might not be the right thing to analyze, because we cannot distinguish, whether it's when user wanted to close the app or because the app is terminated while paused. And then I realized that SDL_APP_TERMINATING event is the one.

I've added some more logs and saw:
```
05-02 20:06:02.964  7689  7689 V SDL     : onDestroy()
05-02 20:06:02.968  7689  7689 I python  : [INFO   ] [window_sdl2.py] _event_filter: action = 'app_terminating'
05-02 20:06:02.969  7689  7717 I python  : [INFO   ] [window_sdl2.py] mainloop: _pause_loop: wait_event done
05-02 20:06:02.970  7689  7717 I python  : [INFO   ] [window_sdl2.py] mainloop: _pause_loop: invoking self._win.poll()
05-02 20:06:02.970  7689  7717 I python  : [INFO   ] [window_sdl2.py] mainloop: _pause_loop: self._win.poll() done, event = ('quit',)
05-02 20:06:02.970  7689  7717 I python  : [INFO   ] [window_sdl2.py] mainloop: _pause_loop: invoking wait_event
```

And that's what I based my second commit on. I'm checking whether the `EventLoop.quit` was set by `_event_filter`. Not there should be no inconsistencies regarding quit event.